src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -270,14 +270,17 @@ where
 
     /// Add kv pair to sstable.
     pub async fn add_monotonic_delete(&mut self, event: MonotonicDeleteEvent) -> HummockResult<()> {
-        if let Some(builder) = self.current_builder.as_mut() && builder.reach_capacity() && event.new_epoch != HummockEpoch::MAX {
-            if builder.last_range_tombstone_epoch() != HummockEpoch::MAX {
+        if let Some(builder) = self.current_builder.as_mut() && builder.reach_capacity() &&
+            && event.new_epoch != HummockEpoch::MAX {
+            if builder.last_range_tombstone_epoch() == HummockEpoch::MAX {
```

Comment:
When this condition is met, will it be guaranteed that `event.event_key.is_exclude_left_key` is false?
- If yes, then we should move this if condition inside `if !event.event_key.is_exclude_left_key`.
- If no, then I think there still can be a corner case unhandled.

Reply:
This code has been changed